### ISSUE TITLE
Geometry.equals and Geometry.equalsExact have inconsistent results for empty geometries

### ISSUE THREAD

**Issue by FObermaier** [locationtech/jts#864]
I noticed that `Geometry.equals` and `Geometry.equalsExact` return different results for empty geometries:
```java
GeometryFactory gf = new GeometryFactory();
Geometry g0 = gf.createPoint();
Geometry g1 = gf.createPoint();

assertEquals(g0.equals(g1), g0.equalsExact(g1));
```
This is due to the fact that the `RelateOp.relate` returns DE9-IM `FFFFFFFF2` instead of `TFFFFFFF2`.

**Comment by dr-jts** [comment]
Yes, this is an anomaly.  Unfortunately the OGC and ISO specs don't offer any guidance here.  They either have no precise definition of `Equals` (e.g. OGC 99-049), or the definition is inconsistent!  (OGC 06-104r4 - see extract below, where the first part implies that "EMPTY = EMPTY", whereas the second part specifies that `I(A) n I(B) <> E`.  I also note that the second definition does not allow `POINT = POINT` , since it requires `B(A) n B(B) <> E`, and Points have no boundary!).

![image](https://user-images.githubusercontent.com/3529053/166265071-0e0882d9-27e2-4cb6-814b-e5247a6b51d3.png)

As you say, changing the semantic to allow empty geometries of same type to match is easy.  But should this be done? 
 
Arguments FOR:
* consistency with `equalsExact`
* seems like the "expected" semantic

Arguments AGAINST:
* no support from OGC/ISO ?
* change in long-standing JTS semantic

The other option is to leave this as it stands, and document this behaviour.

What do other independent systems do?  I don't have any others that I can compare against.

**Comment by dr-jts** [comment]
As a data point, PostGIS semantics are that all empty geometries are **topologically** equal, but only **exactly** equal when of same type.

**Comment by FObermaier** [comment]
SQLServer has the same semantics for STEquals as PostGis: Empty is equal to Empty regardless of the geometry type.
![grafik](https://user-images.githubusercontent.com/2065346/166410311-6b107c1a-60fb-4ce2-98a0-9717caa3a788.png)

**Comment by dr-jts** [comment]
> SQLServer has the same semantics for STEquals as PostGis: Empty is equal to Empty regardless of the geometry type. 

Interesting.  Of course, it may just have copied PostGIS!

Annoying that the standards are so unhelpful about semantics of empty geometries.  

I am leaning towards matching PostGIS/GEOS/SQL Server.  It seems justifiable that "empty is topologically equal".  And this is consistent with the DE-9IM, since `relate(EMPTY, *) = FFFFFFFF2`.

So the definition becomes:
```
equal(A, B) :: FFFFFFFF2
     or dim(A) = dim(B) and (dim(A = 0) -> TFFFFFFF2 or dim(A) > 0  -> TFFFTFFF2)
```
Looks a bit complicated, but I don't see a simpler definition.

This can be implemented by changing `IntersectionMatrix.isEquals`.

(Side note: I wish I'd used `Geometry.equal` and `IntersectionMatrix.isEqual` - that would have avoided some aliasing problems).
